tmp/script_linac.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python-sirius
+"""."""
+
+import time as _time
+from epics import PV as _PV
+import numpy as np
+from optimization import PSO
+
+
+class PSOLinac(PSO):
+
+    def __init__(self, low_lim, up_lim):
+        self._upper_limits = up_lim
+        self._lower_limits = low_lim
+
+    def initialization(self):
+        _pv_phase_shb = _PV('LA-RF:LLRF:BUN1:SET_PHASE')
+        _pv_phase_kly1 = _PV('LA-RF:LLRF:KLY1:SET_PHASE')
+        _pv_phase_kly2 = _PV('LA-RF:LLRF:KLY2:SET_PHASE')
+
+        _pv_amp_shb = _PV('LA-RF:LLRF:BUN1:SET_AMP')
+        _pv_amp_kly1 = _PV('LA-RF:LLRF:KLY1:SET_AMP')
+        _pv_amp_kly2 = _PV('LA-RF:LLRF:KLY2:SET_AMP')
+
+        self.params = [_pv_phase_shb,
+                       _pv_amp_shb,
+                       _pv_phase_kly1,
+                       _pv_amp_kly1,
+                       _pv_phase_kly2,
+                       _pv_amp_kly2]
+
+        _pv_energy = _PV('Linac_Energy')  # Fake PV
+        _pv_spread = _PV('Linac_Energy_Spread')  # Fake PV
+        _pv_ict1 = _PV('ICT1_charge')  # Fake PV
+        _pv_ict2 = _PV('ICT2_charge')  # Fake PV
+
+        self.diag = [_pv_energy,
+                     _pv_spread,
+                     _pv_ict1,
+                     _pv_ict2]
+
+        self._wait = 0
+
+        self.p_energy = 1
+        self.p_spread = 75
+        self.p_transmit = 150
+
+        self._max_delta = self._upper_limits/10
+        self._min_delta = - self._max_delta
+
+        self._nswarm = 10 + 2 * int(np.sqrt(len(self._upper_limits)))
+
+    def calc_merit_function(self):
+        f_out = np.zeros(self._nswarm)
+
+        for i in range(0, self._nswarm):
+            for k in range(0, len(self.params)):
+                self.params[k].value = self._position[i, k]
+
+            _time.sleep(self._wait)
+            transmit = self.diag[3] / self.diag[2]
```

Comment:
It is clearer if you name your indices. Define at the beginning of the class the names you want:
```
ENERGY = 0
SPREAD = 1
ICT1_CHG = 2
ICT2_CHG = 3
```
and then use these definitions here.

Reply:
Nice, thank you!